### DISCUSSION TITLE
How come this isn't just a simple desktop app? Why the need to self-host?

### DISCUSSION THREAD

**Discussion by MMachado05** [Stirling-Tools/Stirling-PDF#1218] [Q&A] [answered]
I've been trying to find open source alternatives to as many proprietary apps as I can, and this came up as one such program for Adobe Acrobat. However, it seems that, in order to use it, there's a *significant* amount of technological overhead you need to get it running. That is, you need to set it up on some external server (maybe) and then either run it in a container or pretty much compile it yourself (seemingly definitely).

Why isn't this packaged as just a simple app? I don't see why a PDF editing and viewing tool should need so much technological know-how to be well-used. I feel that this just adds a significant barrier of entry for those who are less technologically inclined. Are there any plans (if enough funding were to be received) to turn this into a stand-alone, relatively easy-to-install desktop application, much like Acrobat?

Any responses are appreciated!

**Comment by jobritz** [comment]
If you have many Users it has a few advantages with running it as a web-application. You can run updates by updating a single instance and from my point of view it is much faster as a web-application than some complex desktop applications. You can also make it accessible to many different clients without the need of creating a different installation for every operating system

**Comment by Frooodle** [comment, marked as answer]
So few things 

You don't need to compile it yourself
All release artifacts are attached to each release
A .jar and .exe if you wish to run it standalone 

The reason we recommend docker for this app, is that when standalone it's functionality is reduced.
If installed with docker it actually installs several apps that work with Stirling pdf to enhance its functionality (such as adding docx to pdf conversions etc) 


If you were to download the .exe today and run it 
You would get a website opening up with our "ultra-lite" version which has 80% of the full functionality but mises some advanced features 


We have thought about having the .exe version open up a app not web browser (there are tools to do this such as 'electron') 
But it's simply a funding and time issue

**Reply by MMachado05** [reply]
This is very useful info! Thanks for being so thorough!

I suppose my question may have also been born out of an unfamiliarity with docker; would you theoretically be able to install it locally in a container without the need of a home server or anything? A video about this app I found online used a home server, so I just assumed that was the recommended way of doing it.

> But it's simply a funding and time issue

As are most things with open source, it would seem... unfortunate, but something we can all work on together! I'll spread the word about your app and hopefully you'll be able to get some more cashflow to aid in the development process.

Cheers! 😁

**Reply by Frooodle** [reply]
I run "docker desktop" on my home windows PC and its what I use for development of Stirling-PDF as well..
But.. i admit i also have a home server, its certainly not user friendly way compared to .exe etc and i would never recommend it for non tech people

**Reply by MMachado05** [reply]
>...i would never recommend it for non tech people

Hence, why I really hope y'all find the time and funding to develop some native apps! Everyone deserves the ability to choose non-proprietary software, and the work y'all are doing in that avenue is admirable.
